temoa/extensions/stochastics/scenario_creator.py

```
@@ -0,0 +1,101 @@
+    with sqlite3.connect(temoa_config.input_database) as con:
+        hybrid_loader = HybridLoader(db_connection=con, config=temoa_config)
+        data_dict = hybrid_loader.create_data_dict(myopic_index=None)
+
+    # 2. Apply perturbations for this scenario
+    for p in stoch_config.perturbations:
+        if p.scenario != scenario_name:
+            continue
+
+        target_param = data_dict.get(p.table)
+        if target_param is None:
+            logger.warning(f"Table {p.table} not found in data_dict for scenario {scenario_name}")
```

Ruff (0.14.10)

Logging statement uses f-string

(G004)